.github/workflows/ci-build.yaml

```diff
@@ -119,9 +119,11 @@ jobs:
       - name: Upload coverage report
         # engineers just ignore this in PRs, so lets not even run it
         if: github.ref == 'refs/heads/main'
-        uses: codecov/codecov-action@ab904c41d6ece82784817410c45d8b8c02684457 # v3.1.6
+        uses: codecov/codecov-action@84508663e988701840491b86de86b666e8a86bed # v4.3.0
```

Comment:
there's an issue open about 4.3.0 not working either: https://github.com/codecov/codecov-action/issues/1368

but I guess it worked for us 🤷